public/javascripts/data.js

```diff
@@ -219,41 +219,72 @@ var dataNS = odkmaker.namespace.load('odkmaker.data');
     };
 
     /**
-     * Push a translated extra node to a given array.
+     * Push translated extra node to a given array.
+     * 
+     * Nodes are only created if the extra has at least one non-empty string.
+     * 
+     * E.g. `ext` for labels with no extras is an empty object `{}` and will not create any nodes.
+     * 
+     * `ext` for labels with additional, missing translations is an object with some empty or even 
+     * missing keys (languageCodes), e.g. `{0: "English label", 1: ""}` or `{0: "English label"}`. 
+     * Here, langugeCode 1 will get an empty node in both cases.
+     * 
      * @param {Object} arr  The array to be added to.
      * @param {Object} ext  The value of an item of the extra object.
      * @param {Object} frm  The value for "form".
      * @param {String} txn  The desired language.
      * @param {String} pre  An optional prefix for the translated value, e.g. `jr://images/`.
      * @return              None, the arr is mutated in place.
      */
-    var pushChildren = function(arr, ext, frm, txn, pre)
-    {
-        var pre = (typeof pre !== 'undefined') ? pre : "";
-        // Only create a node for non-empty control object values
-        if ((ext !== undefined) && !_.isEmpty(ext[txn._languageCode]))
-        {
+    var pushChildren = function (arr, ext, frm, txn, pre) {
+        var pre = (pre !== undefined) ? pre : "";
+
+        // #268: skip unless ext contains at least one non-empty value
```

Comment:
We test whether e.g. English (first lang) has a value given, while other added languages are empty. In that case, we'll want empty nodes. Only if nothing is given, we can skip creating nodes.